### ISSUE TITLE
[Feature]: Dynamic Kanban board - dataview style

### ISSUE THREAD

**Issue by davidolrik** [mgmeyers/obsidian-kanban#345]
### Goal or desired outcome of this feature

Primary goal: Less manual work, more automation.

Today I have a lot of individual kanban boards, and this makes it hard for me to stay on top of everything.
Adding new tasks and linking them to notes is a bit too much work - i.e. too many manual steps.

### Describe the feature

What I would like to propose is a fenced kanban markdown block that has a dataview query that selects notes for cards and describes the parameters for the kanban board.

- The query will select notes based upon tags or path for the board, just like in dataview
- The kanban board configuration, like columns, lane width, etc. will determine how the board looks
- Meta data in each note will be used for status, due date, completion date, etc.
- Moving cards will update the meta data in the notes
- Adding new cards would only require creating a note that matches the query, which means just adding a note with the matching tags would put it on the board
- Having the kanban board as a fenced markdown block would allow for content before and after the board, making it feel more like the rest of my notes

#### Example

```
``kanban
query: from "Projects" and #task
lanes: ["Todo", "Blocked", "In progress", "Done"]
done: ["Done"]
``
```

This block would create a board with four lanes, with all the notes in the `Projects` folder which are tagged with `#task`

### Can you think of any alternatives or work-arounds?

This feature would probably benefit from a collaboration between this project and the dataview project.

**Comment by SenseiBas** [comment]
This would be GAME CHANGING!!!
If this would integrate with the meta edit plugin you would have a full productivity system!

**Comment by AB1908** [comment]
BTW, I forgot to mention here that now that we have callouts and stuff like Obsidian Columns and cardboard, we can actually get a limited version of this by writing out three different queries and then populating the columns.

**Comment by craldaz** [comment]
I didn't know this discussion already existed. I made a duplicate request here: 
https://github.com/mgmeyers/obsidian-kanban/discussions/558

I've somewhat refined what I proposed in that discussion. I now have a homepage which contains a Kanban board.

![image](https://user-images.githubusercontent.com/19959249/175435961-806aa20c-c6d7-40b8-bf48-ca061b923b56.png)

Cards are added in the Kanban cards note which has several buttons
![image](https://user-images.githubusercontent.com/19959249/175436201-d9b93b07-4285-4fdf-b0d8-9f3de3837b8d.png)

Ideally, as was already mentioned, I'd be able to drag the cards around in the Kanban board and it would update the metadata

**Comment by Levalicious** [comment]
So I'm not sure if this is what was meant when this issue was made, but this is what i've done on my setup: https://gist.github.com/Levalicious/3e179a4b07b127c99e331954365f17b9 

I define a template for cards per board, and then when I use the kanban interface to generate a card, instead of the board scanning for information from cards, the cards scan and parse the board for information.

To adapt the snippet for your own use, if you look, in the top block there's a "path/to/kanban/file" that you'll need to replace, and an equivalent one in the bottom block. The bottom block also has a "kanban/file > " in the middle of it, replace kanban/file with the name (not the path) of your file: make sure to leave the caret and the spaces around it, otherwise you'll pick up some of kanban's formatting.

At the moment I only extract the tags on cards and the name of the list that a card belongs to, but I imagine that extending this wouldn't be too hard.

Try not to judge the code too much, it's a lil icky and I haven't used dataview before today.

**Comment by Xananax** [comment]
There's https://github.com/roovo/obsidian-card-board which does this, but cards can't be dragged around, and it cannot do what Kanban does (simple one file markdown with tasks).

I use both for different usages. Kanban for per-project tasks, cardboard for more generic sparse tasks like homekeeping and such.

I would love to see both functionalities merged, but also I think there's value in having both plugins synergize instead; if both can be configured to use the same date formats and such, maybe even the same column UI, then each plugin can focus on what they do best without getting into spirals of complexity.

**Comment by ValentinBr** [comment]
> There's https://github.com/roovo/obsidian-card-board which does this, but cards can't be dragged around, and it cannot do what Kanban does (simple one file markdown with tasks).
> 
> I use both for different usages. Kanban for per-project tasks, cardboard for more generic sparse tasks like homekeeping and such.
> 
> I would love to see both functionalities merged, but also I think there's value in having both plugins synergize instead; if both can be configured to use the same date formats and such, maybe even the same column UI, then each plugin can focus on what they do best without getting into spirals of complexity.

Yes! Please merge https://github.com/roovo/obsidian-card-board into Kanban. It would give it so much power.

**Comment by ArneNostitz** [comment]
any progress on this?

it would be amazing to have a query, that finds all tasks on project related pages and feeds it into an inbox kanban column - and then be able to drag these tasks out to other columns (somehow maybe tagging them with the column name, so that the inbox knows what is already dragged out?)

So a new task would get "#inbox", dragging it out to column "Finance" would tag it "#finance".

then there would need to be a "modifier+drag" option to keep tags like "#finance" when it gets dragged to a "#doing" or "#done" column. 

- I use a Project overview Kanban to sort depending on field (finance, marketing, dev, ...) and then a second kanban to see what is worked on right now like a status. 

maybe there a better ways - but this is how i do it right now. doing this all with tags could get messy thou and the card-board plugin does not allow for dragging or inline editing of tasks - so its always a going back and forth with files ...

**Comment by serpro69** [comment]
Subscribing to this, sounds like a great addition to an already great plugin.

**Comment by chraaroe** [comment]
Love Obsidian! I became a Sponsor to support Obsidian and show my support for this particular issue. I urge others to do the same.

**Comment by KristianMischke** [comment]
+1 interested in this feature, are there any progress updates here?

**Comment by jteich99** [comment]
+1 on these feature. Please add it!!!

**Comment by AB1908** [comment]
Can you check out the Obsidian Projects plugin meanwhile?

**Comment by lightningRalf** [comment]
With the ability of having dataview queries inside the card. would this be now also possible? Maybe with dataviewjs query at first?

**Comment by alexander-meyenberg** [comment]
This addition would be a game changer. I have spent countless hours trying to cobble together something that will do this and keep failing.
